### PR TITLE
Add option to change kiai glow strength

### LINKED ISSUE #21569: Kiai pulses in osu!taiko are too aggressive
### Type

Game behaviour

### Bug description

Currently, the kiai pulses in taiko are too intense to the point where it goes from annoying/hassle to read on more normal BPM, to actually making eyes hurt, and posing a very concerning risk for epileptic players.
I've used argon skin for these examples but the same issue applies to all skins.

here's an example: https://osu.ppy.sh/beatmapsets/1895673#taiko/3906737

## lazer
https://user-images.githubusercontent.com/62819481/206293976-9d293a2f-0395-44e1-8f11-9c253435a290.mp4

## stable
https://user-images.githubusercontent.com/62819481/206294659-14b71737-438a-4391-abb9-c47ebb5bd73f.mp4

here's a more extreme example: https://osu.ppy.sh/beatmapsets/1732016#taiko/3540250

## lazer
https://user-images.githubusercontent.com/62819481/206294839-48099fa0-9771-4dd5-a99b-69b281fc9f72.mp4

## stable

https://user-images.githubusercontent.com/62819481/206295232-2897e3c3-e8ce-48d5-8edd-563698cca670.mp4

As you can see, the pulses in stable are much more subtle, to the point they're barely noticeable, which makes playing actually bearable. i think adjusting the current intensity of the pulses is a very needed change as of now, because frankly it makes taiko unbearable to play most of the time.

### Screenshots or videos

_No response_

### Version

2022.1207.0-lazer

### Logs

N/A

### LINKED ISSUE #21747: Kiai flashes in osu! ruleset are way too strong
### Type

Cosmetic

### Bug description

Same as #21569 but in osu! ruleset.

It gets really distracting when using skins with colorful hitcircles. Stable has the flashing too, but its way more subtle.

### Screenshots or videos

_No response_

### Version

2022.1208.0

### Logs

-

### PR DESCRIPTION
**Pull request by MaxKruse**
* Fixes #21747
* Fixes #21569 

This adds an option-slider from 10-100% (as suggested by peppy https://github.com/ppy/osu/pull/22015#issuecomment-1369320678 ), although higher values do look a little silly.

As far as I could tell, both Argon skins entirely circumvents kiai glow, Triangles and Legacy (2013) are affected by `KiaiFlash`, and custom user skins fall under `KiaiFlashingDrawable`.

I am unsure on how to add tests for this, manually verifying behaviour in osu!Standard and osu!Taiko seems to work. Other gamemodes did not show any signs of kiai-related visual changes, but that might be because of a lack of experience.

I've added the pre-map option panel as well since this might be feasable to change glow strength on-the-fly rather than going into the overall settings page.

### CODE REVIEW

**Comment by peppy** [comment]
> I've added the pre-map option panel as well since this might be feasible to change glow strength on-the-fly rather than going into the overall settings page.

I thought I mentioned this in discord, but please don't add it there. Settings on that screen are the ones which people would be changing per-beatmap. This is not one of them.

File: osu.Game.Rulesets.Osu/Skinning/Default/KiaiFlash.cs [hunk 1]
```diff
@@ -15,7 +19,8 @@ public partial class KiaiFlash : BeatSyncedContainer
     {
         private const double fade_length = 80;
 
-        private const float flash_opacity = 0.25f;
+        private Bindable<float> flashOpacity = new Bindable<float>();
+        private float opacityValue => flashOpacity.Value;
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Skinning/Default/KiaiFlash.cs:23]: Unnecessary, just refer to this inline.

File: osu.Game.Rulesets.Osu/Skinning/Default/KiaiFlash.cs [hunk 2]
```diff
@@ -15,7 +19,8 @@ public partial class KiaiFlash : BeatSyncedContainer
     {
         private const double fade_length = 80;
 
-        private const float flash_opacity = 0.25f;
+        private Bindable<float> flashOpacity = new Bindable<float>();
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Skinning/Default/KiaiFlash.cs:22]: should be `readonly`. please setup your IDE to give you proper inspections.

File: osu.Game.Rulesets.Osu/Skinning/Default/KiaiFlash.cs [hunk 3]
```diff
@@ -30,13 +35,19 @@ public KiaiFlash()
             };
         }
 
+        [BackgroundDependencyLoader]
+        private void load(OsuConfigManager config)
+        {
+            flashOpacity = config.GetBindable<float>(OsuSetting.GlowStrength);
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Skinning/Default/KiaiFlash.cs:41]: Should not be reassigning to the value. Use `config.BindWith` instead (or don't initialise the value until here).

File: osu.Game/Configuration/OsuConfigManager.cs [hunk 4]
```diff
@@ -161,6 +161,7 @@ protected override void InitialiseDefaults()
             SetDefault(OsuSetting.ScalingPositionY, 0.5f, 0f, 1f);
 
             SetDefault(OsuSetting.UIScale, 1f, 0.8f, 1.6f, 0.01f);
+            SetDefault(OsuSetting.GlowStrength, 0.15f, 0.10f, 1.0f, 0.01f);
```
Comment (peppy) [inline osu.Game/Configuration/OsuConfigManager.cs:164]: Rename to `KiaiGlowStrength`

File: osu.Game/Localisation/GraphicsSettingsStrings.cs [hunk 5]
```diff
@@ -144,6 +144,16 @@ public static class GraphicsSettingsStrings
...
+        /// <summary>
+        /// "Kiai"
+        /// </summary>
+        public static LocalisableString KiaiHeader => new TranslatableString(getKey(@"kiai_header"), @"Kiai");
+
+        /// <summary>
+        /// "Glow Strength"
+        /// </summary>
+        public static LocalisableString GlowStrength => new TranslatableString(getKey(@"kiai_glow"), @"Glow Strength");
+
```
Comment (peppy) [inline osu.Game/Localisation/GraphicsSettingsStrings.cs:156]: Combine these two. There's not going to be any other kiai settings.

Comment (peppy) [reply osu.Game/Localisation/GraphicsSettingsStrings.cs:156]: Also this should likely be under gameplay, not grapihcs.

**Review by peppy — CHANGES_REQUESTED** [review]
Quick initial review, much to fix.

**Comment by Joppe27** [comment]
> As far as I could tell, both Argon skins entirely circumvents kiai glow

This is not the case for taiko:

https://github.com/ppy/osu/blob/c2599957661692a423ab6b3bda04c4b9942d350e/osu.Game.Rulesets.Taiko/Skinning/Argon/ArgonCirclePiece.cs#L102-L114

I also suggest calling this setting "kiai flash" instead of "kiai glow" to avoid confusion with `taiko-glow`, which is an unrelated element that is also only displayed during kiai time.

**Comment by peppy** [comment]
Flash also sounds fine.

File: osu.Game.Rulesets.Osu/Skinning/Default/KiaiFlash.cs (hunk 2 above)
Comment (MaxKruse) [reply osu.Game.Rulesets.Osu/Skinning/Default/KiaiFlash.cs]: Somehow doesnt show up in VS2022. Will investigate inspections for the future.

File: osu.Game.Rulesets.Osu/Skinning/Default/KiaiFlash.cs (hunk 3 above)
Comment (MaxKruse) [reply osu.Game.Rulesets.Osu/Skinning/Default/KiaiFlash.cs]: Was unaware of this method to bind 👍

File: osu.Game/Configuration/OsuConfigManager.cs (hunk 4 above)
Comment (MaxKruse) [reply osu.Game/Configuration/OsuConfigManager.cs]: Renamed to KiaiFlash in all places as suggested below.

**Comment by RaiLTC** [comment]
I would like to suggest allowing to completely disable kiai flashes in Taiko, reason being is I'm visually impaired with less than 10 degree vision (peripheral) in both of my eyes, and slight change in color messes up my ability to read patterns on higher level difficulties. And disabling the kiai completely will also allow me to play in the future as my vision gets worse with time (there is no treatment available for Retinitis Pigmentosa)

**Comment by peppy** [comment]
I've fixed this PR up, but on testing I think it feels objectively bad to have kiai lower than the default.

Probably won't merge this one. Will wait for more feedback.

**Comment by zethz161** [comment]
hello
i would support having this option (both the ability to toggle it off completely, and adjusting the intensity) available for lazer (especially for taiko, where i spend most of my time on). given how some intense storyboards are required to have the epilepsy warning turned on in the editor, it would also make sense to consider this option for normal gameplay given that 180bpm kiai flashes is already enough to trigger photosensitive epilepsy (minimum 3hz of flashes, 3 beats every second = 180bpm), and turning on the epilepsy warning for every map >180bpm just wouldnt make sense
i personally think the flashes on argon are really intense, and while it doesnt bother me, it would be problematic to some players who are sensitive to lights and its probably a good idea to consider them as well when making this change

**Comment by peppy** [comment]
Closing for now, we can re-discuss in the future. If you notice any kiai effects being brighter than stable, please open / bump relevant issues.

**Comment by ArijanJ** [comment]
@peppy I'm really not sure why you are so against having the option to remove, or at least diminish, a purely visual feature which multiple people have reported to cause eye strain/issues with eye conditions.

One of the comments in this thread has 13 thumbs up, as well as several reddit threads and [forum posts](https://osu.ppy.sh/community/forums/topics/1722858) complaining about the same.

I would be happy to rebase this onto master if you ever change your mind, as it only increases accessibility with no downside at all. FWIW I mainly have an issue with this in taiko, when playing a map for multiple minutes on a black background and the sudden flashing completely takes me out of the zone and makes it much more difficult to read patterns.

**Pull request closed without merging**